bin/csv_from_oai.rb

```diff
@@ -0,0 +1,65 @@
+
+require_relative "../lib/oai/client_decorator"
+
+# Input for dynamic email
+puts "Enter your email address:"
+email = gets.chomp
+
+# OAI client setup
+client = OAI::Client.new(
+  "http://oai.adventistdigitallibrary.org/OAI-script",
+  headers: { from: email },
+  parser: 'libxml'
+)
+
+opts = {
+  metadata_prefix: "oai_adl"
+}
+
+CSV.open('csv_from_oai.csv', 'wb') do |csv|
+  # Write the headers to the CSV file
+  csv << ['set', 'identifier', 'derivative_type', 'URL']
+
+  sets = [
+    "adl:thesis",
+    "adl:periodical",
+    "adl:issue",
+    "adl:article",
+    "adl:image",
+    "adl:book",
+    "adl:other"
+  ]
+
+  sets.each do |set|
+    records = client.list_records(opts.merge(set: set))
+    # For the first 25 records
+    records.each_with_index do |r|
+    # For all records, comment out previous line and comment in the following.
+    # records.full.each_with_index do |r|
+      identifier = r.header.identifier
+      thumbnail_urls = r.metadata.first.find('thumbnail_url').map(&:content)
+      related_urls = r.metadata.first.find('related_url').map(&:content)
+
+      thumbnail_urls.each do |thumbnail_url|
+        derivative_type = "thumbnail"
+        if thumbnail_file_type = thumbnail_url.split(/[.;]/).last
```

Comment:
What is the text of the `related_url` at this point?  

I see that you are splitting on the regular expression `/[.;]/` which is very similar to splitting on `;`.  But somewhat more confusing is that you switched from using the `first` value from the split and are now using the `last` value.  Why the switch from `first` to `last`?